src/pages/MovieDetails.tsx

```diff
@@ -0,0 +1,101 @@
+// pages/MovieDetails.tsx
+// import React from 'react';
+import { useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { useQuery } from '@tanstack/react-query';
+import { Movie } from '../Components/MovieCard';
+import { Switch } from '@/Components/ui/switch';
+
+const fetchMovieDetails = async (id: string, language: string) => {
+  const response = await fetch(
+    `https://api.themoviedb.org/3/movie/${id}?api_key=${
+      import.meta.env.VITE_TMDB_API_KEY
+    }&language=${language}`
+  );
+  if (!response.ok) {
+    throw new Error('Failed to fetch movie details');
+  }
+  return response.json();
+};
+
+const MovieDetails = () => {
+  const { id } = useParams<{ id: string }>();
+  const [language, setLanguage] = useState('en');
+  const handleLanguageChange = () => {
+    setLanguage((prevLanguage) => (prevLanguage === 'en' ? 'ru' : 'en'));
+  };
+
+  const {
+    data: movie,
+    isLoading,
+    error,
+  } = useQuery<Movie>({
+    queryKey: ['movie', id, language],
```

Comment:
```suggestion
    queryKey: ['movie', id, { language }],
```

правильнее складывать фильтры и доп вещи в объект

Reply:
fixed